autoload/intero/process.vim

```diff
@@ -376,3 +302,94 @@ function! s:build_complete(job_id, data, event) abort
         endif
     endif
 endfunction
+
+function! s:ensure_intero_is_installed() abort
+    if exists('g:intero_built')
+        return
+    endif
+
+    " If `stack` exits with a non-0 exit code, that means it failed to find the executable.
+    if (!executable('stack'))
+        echom 'Stack is required for Intero. Aborting.'
+        return
+    endif
+
+    " We haven't set the stack-root yet, so we shouldn't be able to find this yet.
+    if (executable('intero'))
+        echom 'Intero is installed in your PATH, which may cause problems when using different resolvers.'
+        echom 'This usually happens if you run `stack install intero` instead of `stack build intero`.'
+        echom 'Aborting.'
+        return
+    endif
+
+    if g:intero_use_neomake && !exists(':Neomake')
+        echom 'Neomake not detected. Flychecking will be disabled.'
+    endif
```

Comment:
You should probably move this check outside of this function, since it applies regardless of the backend we're using.

Reply:
Good catch. I meant to ask about that, wasn't sure.